skrub/_text_encoder.py

```diff
@@ -0,0 +1,405 @@
+
+        - ``intfloat/e5-small-v2``
+        - ``all-MiniLM-L6-v2``
+        - ``all-mpnet-base-v2``
+
+        You can find more options on the `sentence-transformers documentation
+        <https://www.sbert.net/docs/pretrained_models.html#model-overview>`_.
+
+        The default model is a shrinked version of e5-v2, which has shown good
+        performance in the benchmark of [1]_.
+
+    n_components : int or None, default=30,
+        The number of embedding dimensions. As the number of dimensions is different
+        accross embedding models, this class uses a :class:`~sklearn.decomposition.PCA`
+        to set the number of embedding to ``n_components`` during ``transform``.
+        Set ``n_components=None`` to skip the PCA dimension reduction mechanism.
+
+        See [1]_ for more details on the choice of the PCA and default
+        ``n_components``.
+
+    device : str, default=None
+        Device (e.g. "cpu", "cuda", "mps") that should be used for computation.
+        If None, checks if a GPU can be used.
+        Note that macOS ARM64 users can enable the GPU on their local machine
+        by setting ``device="mps"``.
+
+    batch_size : int, default=32
+        The batch size to use during ``transform``.
+
+    token_env_variable : str, default=None
+        The name of the environment variable which stores your HuggingFace
+        authentication token to download private models.
+        Note that we only store the name of the variable but not the token itself.
+
+    cache_folder : str, default=None
+        Path to store models. By default ``~/skrub_data``.
+        See :func:`skrub.datasets._utils.get_data_dir`.
+        Note that when unpickling ``TextEncoder`` on another machine,
+        the ``cache_folder`` path needs to be accessible to store the downloaded model.
+
+    store_weights_in_pickle : bool, default=False
+        Whether or not to keep the loaded sentence-transformer model
+        in the TextEncoder when pickling.
+
+        - When set to False, the ``_estimator`` property is removed from
+          the objects to pickle, which significantly reduces the size of
+          the serialized object. Note that when the serialized object is
+          unpickled on another machine, the TextEncoder will try to download
+          the sentence-transformer model again from HuggingFace Hub.
+          This process could fail if, for example, the machine doesn't have
+          internet access. Additionally, if you use weights stored on disk
+          that are *not* on the HuggingFace Hub (by passing a path to
+          ``model_name``), these weights will not be pickled either.
+          Therefore you would need to copy them to the machine where you
+          unpickle the ``TextEncoder``.
+        - When set to True, the ``_estimator`` property is included in
+          the serialized object. Users deploying fine-tuned models stored on
+          disk are recommended to use this option.
```

Comment:
maybe point out that the machine where we unpickle must have the correct device?